97 crunchbase baseline

Juan-Mateos opened the pull request:
Closes #96 

Follow instructions (https://github.com/nestauk/ai_genomics/tree/97_crunchbase_baseline/ai_genomics/pipeline`ai_genomics/pipeline/README.md")[here] to create a table with AI genomics organisations and AI / genomics baselines.

Includes getter to read the table locally or from s3.

---

Checklist:

- [x] I have refactored my code out from `notebooks/`
- [x] I have checked the code runs
- [ ] I have tested the code
- [x] I have run `pre-commit` and addressed any issues not automatically fixed
- [x] I have merged any new changes from `dev`
- [x] I have documented the code
  - [x] Major functions have docstrings
  - [ ] Appropriate information has been added to `README`s
- [x] I have explained the feature in this PR or (better) in `output/reports/`
- [x] I have requested a code review

Commit:
script to collect data

Commit:
updated getters to fetch crunchbase table

Commit:
added fetch_s3 function

Commit:
added pipeline README

Commit:
changed cb company getter name

india-kerle commented on ai_genomics/getters/crunchbase.py:
```diff
@@ -13,3 +21,17 @@ def get_ai_genomics_crunchbase_org_ids() -> pd.DataFrame:
         raise e
+
+
+def get_ai_genomics_crunchbase_orgs(local: bool = True) -> pd.DataFrame:
```
you can use load_s3_data from ai_genomics.getters.data_getters!

india-kerle commented on ai_genomics/getters/crunchbase.py:
```diff
@@ -1,6 +1,14 @@
+    CB_COMP_NAME,
+)
+
+from ai_genomics.utils.reading import fetch_s3
```
from ai_genomics.getters.data_getters import load_s3_data

india-kerle commented on ai_genomics/pipeline/crunchbase_data/make_crunchbase_data.py:
```diff
@@ -0,0 +1,142 @@
+        return np.nan
+
+
+def send_output_to_s3(file_path: str, s3_destination: str):
```
use save_to_s3 in ai_genomics.getters.data_getters :)

india-kerle commented on ai_genomics/pipeline/crunchbase_data/make_crunchbase_data.py:
```diff
@@ -0,0 +1,142 @@
+        cb_comps[["ai", "genom", "ai_genom"]].values.sum(axis=1) > 0, :
+    ].reset_index(drop=True)[KEEP_CB_COLS].to_csv(CB_COMP_PATH, index=False)
+
+    send_output_to_s3(str(CB_COMP_PATH), f"crunchbase/{CB_COMP_NAME}")
```
we've been saving datasets in inputs instead of outputs in s3 - perhaps that would still be a better location

india-kerle commented on ai_genomics/pipeline/crunchbase_data/make_crunchbase_data.py:
```diff
@@ -0,0 +1,142 @@
+os.makedirs(CB_INPUTS_DATA_DIR, exist_ok=True)
+os.makedirs(CB_OUTPUTS_DATA_DIR, exist_ok=True)
+
+KEEP_CB_COLS = [
```
perhaps better placed in crunchbase utils?

Commit:
changed s3 functions in getters

Commit:
moved column list to keep to utils

Commit:
fixed target directory in s3

Juan-Mateos commented:
Thank you for your review!

I addressed all comments except changing target dirs in S3. One for the future!

I have run all code and getters and everything seems to be fine.

india-kerle commented:
looks good! feel free to merge
